activesupport/lib/active_support/claims.rb

```diff
@@ -0,0 +1,69 @@
+require 'active_support/time'
+
+module ActiveSupport
+  class Claims # :nodoc:
+    class InvalidClaims < StandardError; end
+    class ExpiredClaims < StandardError; end
+    
+    attr_reader :payload, :purpose, :expires_at
+    UNIVERSAL_PURPOSE = 'universal'
+
+    def initialize(payload:, **options)
+      @payload = payload
+      @purpose = self.class.pick_purpose(options)
+      @expires_at = pick_expiration(options)
+    end
+
+    class << self
+      attr_accessor :expires_in
+
+      def verify!(claims, options = {})
```

Comment:
I thought upon it for quite a while @kaspth. I still don't understand why we want to convert what we are doing in ```verify``` and ```verify!``` into ```if``` and ```else``` of ```validate``` :smile: . We would be able to add extra metadata afterwards, and check for them in ```verify``` and ```verify!``` functions. Whatever we do, we have to repeat some code. Be it in ```if``` and ```else``` of ```validate``` or directly in ```verify``` and ```verify!``` methods.

Anyways, I have made things more readable like this. What do you think about this? :

```ruby
      def verify!(claims, options = {})
        raise_if_invalid_purpose(claims[:for], options[:for])
        claims[:pld] if raise_if_expired(claims[:exp])
      end

      def verify(claims, options = {})
        claims[:pld] if check_purpose?(claims[:for], options[:for]) && not_expired?(claims[:exp])
      end

      private
        def not_expired?(expiration)
          return true unless expiration

          Time.iso8601(expiration) > Time.now.utc
        end

        def check_purpose?(claims_purpose, other_purpose)
          claims_purpose == (other_purpose || UNIVERSAL_PURPOSE)
        end

        def raise_if_invalid_purpose(claims_purpose, other_purpose)
          check_purpose?(claims_purpose, other_purpose) || raise(InvalidClaims)
        end

        def raise_if_expired(expiration)
          not_expired?(expiration) || raise(ExpiredClaims)
        end
```